kubernetes/config-maps/nginx-storage/default.conf.j2

```diff
@@ -0,0 +1,96 @@
+server {
+    listen 80;
+
+    gzip                     on;
+    gzip_min_length          1000;
+    gzip_proxied             expired no-cache no-store private auth;
+    gzip_types               text/plain text/css text/javascript application/xml;
+    client_max_body_size     {{ max_upload_size }};
+
+
+    location /download/ {
+        alias /srv/storage/;
+    }
+
+    location = /upload/ {
+
+        access_by_lua_block {
+            -- Module nginx big upload support only content-type: "application/x-www-form-urlencoded"
+            check_content_type_header=ngx.req.get_headers()["Content-Type"]
+            if check_content_type_header ~= "application/x-www-form-urlencoded" then
+                ngx.status=400
+                ngx.say('Only Content-Type: "application/x-www-form-urlencoded" is allowed!')
+                ngx.exit(400)
+            end
+            -- Make sure that header "Concent-Upload-Path" with path and filename was submitted in request
+            check_concent_upload_path_header=ngx.req.get_headers()["Concent-Upload-Path"]
+            if check_concent_upload_path_header == nil then
+                ngx.status=400
+                ngx.say('Header "Concent-Upload-Path" must be submitted!')
+                ngx.exit(400)
+            end
+            -- Make sure that uploaded file has content
+            check_content_length_header=ngx.req.get_headers()["Content-Length"]
+            if check_content_length_header == "0" then
+                ngx.status=400
+                ngx.say('The uploaded file could not be empty!')
+                ngx.exit(400)
+            end
+        }
+
+        proxy_pass http://nginx-storage/upload_internal/;
+        # Make sure that user could not submitted header "X-Session-Id" with content
+        # This header is responsible for temporary filename
+        proxy_set_header X-Session-Id "";
+    }
+
+    location /upload_internal/ {
```

Comment:
Why is there no `internal` directive inside  `/upload-internal/`? It should not be accessible from the outside.

Reply:
Done.

Reply:
Not working with `internal` directive. In nginx documentation, you can see possible internal request. http://nginx.org/en/docs/http/ngx_http_core_module.html#internal.

Reply:
OK so `internal` won't work. But we really can't leave it publicly accessible. Otherwise someone could bypass `/upload/` and send files directly to `/upload-internal/`, possibly using his own value for `X-Session-Id` or skipping gatekeeper checks.

Please try replacing `/upload-internal/` with `@upload-internal`. If that does not work, we'll need to find some other way to make it inaccessible.

Reply:
Ok. So, it's look like solution work, but require change for stripping `X-Session-Id`.